python/rmm/_lib/cuda_stream_view.pyx

```diff
@@ -15,22 +15,27 @@
 import os
 
 from libc.stdint cimport uintptr_t
-from rmm._lib.lib cimport cudaStreamPerThread
+
+
+cpdef enum cudaStream:
+    cudaStreamDefault = 0
+    cudaStreamLegacy = 1
+    cudaStreamPerThread = 2
 
 
 cdef class CudaStreamView:
 
-    def __cinit__(self, uintptr_t stream=0):
+    def __cinit__(self, cudaStream stream=cudaStreamDefault):
```

Comment:
Do you think we should expose an enum in the public API, or accept a string (`"default"  | "legacy" | "per_thread"`) that we translate into an Enum?

Both sound OK to me, but I feel end users are less likely to be familiar with enums?

Reply:
I believe integers are implicitly castable into Python Enums so they can continue to pass an integer as expected.

Reply:
I don't have a strong opinion on either one, I agree that Python users may be more comfortable with a string though, so if we have more people in favor of that I'll change it. Alternatively, we can document that the valid values for those are `rmm._lib.cuda_stream_view.cudaStream{Default,Legacy,PerThread}`, of course, we can probably add that to `__init__.py` so it's more convenient for users to specify just `rmm.cudaStream{Default,Legacy,PerThread}`.

Reply:
Users need the ability to pass an arbitrary stream here as well, so I'm a HUGE -1 to anything involving strings.

Reply:
I'm ok not involving strings, we could still work to make both strings and integers to function properly, but certainly the enum approach is the least convoluted one, so I'm ok with keeping it.